src/modules/emote_menu/components/Sidebar.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {useEffect, useRef} from 'react';
+import Nav from 'rsuite/lib/Nav/index.js';
+import styles from '../styles/sidebar.module.css';
+
+const ITEM_HEIGHT = 42;
+const SIDEBAR_HEIGHT = 300;
+
+let timer;
+
+export default function Sidebar({section, onChange, providers}) {
+  const containerRef = useRef(null);
+
+  useEffect(() => {
+    if (section.eventKey == null) return;
+
+    clearTimeout(timer);
+
+    timer = setTimeout(() => {
+      const top = containerRef.current.scrollTop;
+      const index = providers.findIndex((provider) => provider.id === section.eventKey);
+      const depth = index * ITEM_HEIGHT;
+
+      if (depth < top) {
+        containerRef.current.scrollTo({
+          top: depth,
+          left: 0,
+          behavior: 'smooth',
+        });
+
+        return;
+      }
+
+      if (depth + ITEM_HEIGHT > top + SIDEBAR_HEIGHT) {
+        containerRef.current.scrollTo({
```

Comment:
we wrote this logic twice, but it seems like we instead want to compute `top` as a variable, and pass that for both.

```
let newTop;
if (depth < top) {
  newTop = depth;
}
if (depth + ITEM_HEIGHT > top + SIDEBAR_HEIGHT) {
  newTop = depth - SIDEBAR_HEIGHT + 36; // where does 36 come from? why isnt that a constant?
}
if (newTop == null) {
  return;
}

containerRef.current.scrollTo(...);
```